packages/user_packages/iotcc_mqtt.py

```diff
@@ -79,7 +80,17 @@ def run(self, registry):
             self.iotcc.set_system_properties(self.iotcc_edge_system, registry.get("system_properties"))
             # Set the properties for edge system as key:value pair, you can also set the location
             # by passing the latitude and longitude as a property in the user package
-            self.iotcc.set_properties(self.iotcc_edge_system,{"key1": "value1", "key2": "value2"})
+            # If the set_properties or register call fails due to DCC_Comms Publish exception
+            # the optional retry mechanism can be implemented in the following way
+            attempts = 0
+            while attempts < 3:
+                try:
+                    # Register edge system (gateway)
+                    self.iotcc.set_properties(self.iotcc_edge_system, {"key1": "value1", "key2": "value2"})
+                except Exception:
+                    attempts += 1
+                    # The sleep time before re-trying depends on broker restart
```

Comment:
I don't understand this comment since the sleep time is a constant; and don't we need all of register/set_system_properties/set_properties to succeed to function properly?

Reply:
We have just provided an example to liota developer if required he can change the wait time or even remove it. Added elaborated comments.